packages/tailwindcss/src/candidate.ts

```diff
@@ -557,6 +557,13 @@ export function parseVariant(variant: string, designSystem: DesignSystem): Varia
         case 'compound': {
           if (value === null) return null
 
+          // Forward modifier from `not-*` and `has-*` variants to the
+          // compound variant.
+          if (modifier !== null && (root === 'not' || root === 'has')) {
+            value = `${value}/${modifier}`
+            modifier = null
+          }
+
```

Comment:
This is more correct, but feels so wrong in this spot

Reply:
yeah agree it does feel wrong 🤔

Could in theory add a `forwardsModifier()` method to Variants and control it from there? Gonna think on this some